activerecord/test/cases/connection_adapters/schema_cache_test.rb

```diff
@@ -40,17 +40,22 @@ def test_clearing
       def test_dump_and_load
         @cache.columns('posts')
         @cache.columns_hash('posts')
-        @cache.tables('posts')
+        @cache.data_sources('posts')
         @cache.primary_keys('posts')
 
         @cache = Marshal.load(Marshal.dump(@cache))
 
         assert_equal 11, @cache.columns('posts').size
         assert_equal 11, @cache.columns_hash('posts').size
-        assert @cache.tables('posts')
+        assert @cache.data_sources('posts')
         assert_equal 'id', @cache.primary_keys('posts')
       end
 
+      def test_table_methods_deprecation
+        assert_deprecated { assert @cache.table_exists?('posts') }
```

Comment:
why? I put the `assert` there to make sure that it actually still returns `true` in the case of an existing table.